src/main/webapp/app/shared/icons/MutationConvertIcon.tsx

```diff
@@ -0,0 +1,84 @@
+import { faExchangeAlt } from '@fortawesome/free-solid-svg-icons';
+import React, { useEffect, useState } from 'react';
+import ActionIcon, { IActionIcon } from './ActionIcon';
+import { PRIMARY } from 'app/config/colors';
+import { Mutation } from '../model/firebase/firebase.model';
+import { onValue, ref } from 'firebase/database';
+import { IRootStore } from 'app/stores';
+import { componentInject } from '../util/typed-inject';
+import { observer } from 'mobx-react';
+import { parseAlterationName } from '../util/utils';
+
+export interface IMutationConvertIconProps extends Omit<IActionIcon, 'icon'>, StoreProps {
+  convertTo: 'vus' | 'mutation';
+  mutationName: string;
+  mutationUuid?: string;
+  firebaseMutationsPath?: string;
+}
+
+const MutationConvertIcon = ({
+  firebaseDb,
+  convertTo,
+  firebaseMutationsPath,
+  mutationName,
+  mutationUuid,
+  color,
+  onClick,
+  tooltipProps,
+  ...actionIconProps
+}: IMutationConvertIconProps) => {
+  const [mutationList, setMutationList] = useState<Mutation[]>(null);
+  const [isDisabled, setIsDisabled] = useState(false);
+
+  useEffect(() => {
+    let unsubscribe;
+    if (convertTo === 'vus') {
+      unsubscribe = onValue(ref(firebaseDb, firebaseMutationsPath), snapshot => {
+        setMutationList(snapshot.val());
```

Comment:
This is that expensive listener again. Is it really necessary to listen on the whole mutations?